pkg/controller/statefulset/stateful_pod_control.go

```diff
@@ -241,6 +241,10 @@ func (spc *StatefulPodControl) UpdatePodClaimForRetentionPolicy(ctx context.Cont
 			logger.V(4).Info("Expected claim missing, continuing to pick up in next iteration", "PVC", klog.KObj(claim))
 		case err != nil:
 			return fmt.Errorf("Could not retrieve claim %s not found for %s when checking PVC deletion policy: %w", claimName, pod.Name, err)
+		case hasUnexpectedController(claim, set, pod):
+			// Do not set any ownerRefs if there is an unexpected controller.
+			msg := fmt.Sprintf("Claim %s has a conflicting controller, the retention policy is ignored for this claim", claimName)
+			spc.recorder.Event(set, v1.EventTypeWarning, "ConflictingController", msg)
```

Comment:
also this part from the KEP seems not to be satisfied and should apply for all policy combinations: 

> If there is another controller, the existing non-controller StatefulSet or Pod reference will be removed.

Reply:
Thanks for taking a look at this! You make some good points.

Let's resolve the conversation on the KEP first about what the proper semantics should be, then I'll work through your comments here -- you raise some excellent points.

Reply:
argh, you are correct. Thank you.

Reply:
Change is in progress. It was better to make a little more complete re-vamp of the logic to keep it from getting too complicated.

Reply:
Thanks for looking at that. And yeah, there are a lot of cases that need to be covered. It would be nice to have them all covered by the tests.